crates/chia-consensus/src/gen/condition_tools.rs

```diff
@@ -0,0 +1,194 @@
+    #[case(AGG_SIG_ME, b"me_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 1303)]
+    fn test_make_aggsig_final_message(
+        #[case] opcode: ConditionOpcode,
+        #[case] msg: &[u8],
+        #[case] parent_id: Vec<u8>,
+        #[case] puzzle_hash: Vec<u8>,
+        #[case] coin_amount: u64,
+    ) {
+        use std::sync::Arc;
+
+        use chia_protocol::Bytes32;
+
+        use crate::r#gen::conditions::Spend;
+
+        let mut expected_result = Vec::<u8>::new();
+        expected_result.extend(msg);
+
+        let coin = Coin::new(
+            Bytes32::try_from(parent_id.clone()).expect("test should pass"),
+            Bytes32::try_from(puzzle_hash.clone()).expect("test should pass"),
+            coin_amount,
+        );
+
+        match opcode {
+            AGG_SIG_PARENT => {
+                expected_result.extend(parent_id.as_slice());
+                expected_result.extend(TEST_CONSTANTS.agg_sig_parent_additional_data.as_slice());
+            }
+            AGG_SIG_PUZZLE => {
+                expected_result.extend(puzzle_hash.as_slice());
+                expected_result.extend(TEST_CONSTANTS.agg_sig_puzzle_additional_data.as_slice());
+            }
+            AGG_SIG_AMOUNT => {
+                expected_result.extend(u64_to_bytes(coin_amount).as_slice());
+                expected_result.extend(TEST_CONSTANTS.agg_sig_amount_additional_data.as_slice());
+            }
+            AGG_SIG_PUZZLE_AMOUNT => {
+                expected_result.extend(puzzle_hash.as_slice());
+                expected_result.extend(u64_to_bytes(coin_amount).as_slice());
+                expected_result.extend(
+                    TEST_CONSTANTS
+                        .agg_sig_puzzle_amount_additional_data
+                        .as_slice(),
+                );
+            }
+            AGG_SIG_PARENT_AMOUNT => {
+                expected_result.extend(parent_id.as_slice());
+                expected_result.extend(u64_to_bytes(coin_amount).as_slice());
+                expected_result.extend(
+                    TEST_CONSTANTS
+                        .agg_sig_parent_amount_additional_data
+                        .as_slice(),
+                );
+            }
+            AGG_SIG_PARENT_PUZZLE => {
+                expected_result.extend(parent_id.as_slice());
+                expected_result.extend(puzzle_hash.as_slice());
+                expected_result.extend(
+                    TEST_CONSTANTS
+                        .agg_sig_parent_puzzle_additional_data
+                        .as_slice(),
+                );
+            }
+            AGG_SIG_ME => {
+                expected_result.extend(coin.coin_id().as_slice());
+                expected_result.extend(TEST_CONSTANTS.agg_sig_me_additional_data.as_slice());
+            }
+            _ => {}
+        };
```

Comment:
rather than computing the expected value programatically, I think it would be easier to read this test (and be confident it's correct) if the expected output was a parameter instead. Since these are binary strings, it could be simplified by either altering the test constants to make the additional data ascii, or to change the messages to be hex and then specify the expected output as hex. Given that the amount will be binary either way, that might be the better option